pywr/_model.pyx

```diff
@@ -641,7 +649,12 @@ class Model(object):
 
         components = self.flatten_component_tree(rebuild=True)
         for component in components:
-            component.setup()
+            try:
```

Comment:
We could do this in `reset` and `finish` too. Similar to #886 for the other functions such as `before` and `after` which are called every timestep. I.e. we could do it there but it might have performance hit. 

I'm also not sure that this will catch all exceptions because `AbstractNode.setup` is a cpdef function. I assume this helps catch something from setup, but I am guessing it's from Python code not Cython? 

Finally, this change could be done to `master` and ported here if we're happy with it. @Batch21 any thoughts?

Reply:
I'm happy to re-do this into master if requested, or is that something you can do directly?

Reply:
The reason for adding this was that the new ValueError raise load_parameter in parameters.pyx didn't have access to the node / parameter name. So I added this additional catch to log that info. So yes, the original raise came from Cython.

In terms of efficiency, it's OK to add do `finish` and `reset`, right? as these are not called each timestep. If so, i'm happy to add them to the PR.

Reply:
Any objection to me wrapping the `node.setup()` calls in try: except?

Reply:
> I'm happy to re-do this into master if requested, or is that something you can do directly?

I've not looked at your commits. Can these changes be cherry-picked without the marshmallow stuff? Re stuff below it would be good to get these changes in master, so if possible can you separate them?

> So yes, the original raise came from Cython.

OK. I am not sure I fully understand how the exceptions propagate! I think `load_parameter` is just a `def` even though it is a Cython module, so it will propagate correctly. It's for the `.setup` methods which are `cpdef * except -1` which I think will not work. I think we should probably make a test with a custom Node/Parameter that simply create exceptions to make sure this works as expected. However, that's expanding the scope of this PR a bit. 

> In terms of efficiency, it's OK to add do finish and reset, right?

I think so. However, if possible this could be done in a PR to master.

> Any objection to me wrapping the node.setup() calls in try: except?

No, that's probably a good idea. Same caveats to `Parameter.setup()` apply though.

Reply:
The commits can't be cherry picked cleanly as I didn't distinguish between schema and non-schema related stuff. I could create a new branch off master locally, merge, and then do a PR from there?

I've wrapped the node.setup(self) in the reset and setup functions, so hopefully that won't have much of a performance hit. Please let me know if that's a correct assumption or not!

As for extending the tests to except, I'm happy to add that to this MR if requested.

Reply:
Sorry I missed these responses!

> The commits can't be cherry picked cleanly as I didn't distinguish between schema and non-schema related stuff.

It would be nice to have the non-schema stuff in master. Then we can also solve the other stuff (below) there too.

> I've wrapped the node.setup(self) in the reset and setup functions, so hopefully that won't have much of a performance hit. Please let me know if that's a correct assumption or not!

That should be fine.

> As for extending the tests to except, I'm happy to add that to this MR if requested.

We can perhaps tidy this up with #886 in one go.